core-bundle/src/Resources/contao/templates/_new/content_element/toplink.html.twig

```diff
@@ -0,0 +1,36 @@
+{% trans_default_domain "contao_default" %}
+{% extends "@Contao/content_element/_base.html.twig" %}
+
+{% block content %}
+    {% block link %}
+        {% set link_attributes = attrs(link_attributes|default)
+            .set('href', '#')
+            .set('data-toplink')
+            .setIfExists('title', link_text)
+        %}
+        <a{% block link_attributes %}{{ link_attributes }}{% endblock %}>
+            {%- block link_inner %}
+                {{- link_text|default('MSC.backToTop'|trans) -}}
+            {% endblock -%}
+        </a>
+    {% endblock %}
+
+    {% block script %}
+        {% if not as_editor_view %}
+            {% add "toplink_script" to body %}
+                <script type="application/javascript">
+                    {# Scroll toplink_anchor into view when an element with a data-toplink attribute is clicked #}
+                    const toplink_anchor = document.getElementById('{% block toplink_anchor %}top{% endblock %}');
+
+                    document.querySelectorAll('*[data-toplink]').forEach(link => link.addEventListener('click', e => {
+                        e.preventDefault();
+                        toplink_anchor.scrollIntoView({
+                            block: '{% block toplink_scroll_align %}start{% endblock %}',
+                            behavior: '{% block toplink_scroll_behavior %}smooth{% endblock %}',
+                        });
+                    }));
+                </script>
```

Comment:
Do we actually need this script?

IMO, if smooth scrolling is desired, the CSS code `body { scroll-behavior: smooth; }` will be used anyways and no javascript is needed. Just linking to `#` or `#top` is supported by every browser AFAIK.

Reply:
Hmm. Am I missing something? For me (Chrome and Firefox on Linux) a link `<a href="#">top</a>` loads the page `https://my-domain.wip/#` instead. Does this maybe have something to do with the base tag?

Reply:
Yes, I forgot about the base tag 😓

Reply:
Is there a forward compatbile way so that we simply could remove the script once the base tag is gone and nothing else would change? I assume we should not set smooth scrolling and the like then, should we?

Reply:
```suggestion
                <script>
                    document.querySelectorAll('*[data-toplink]').forEach(link => {
                        link.href = location.href;
                        link.hash = '#top';
                    });
                </script>
```

I’d just fix the URL via the JavaScript then. WDYT?

Reply:
Great idea! :heart_eyes: 

Done in 006c4ab3d3f1e70c07d2c1bf91f8e11a3a0af798.